File: documentation/Exercise 10/Java/src/main/java/de/thu/recipebook/runnables/DeleteRecipeRunnable.java
package de.thu.recipebook.runnables;

import android.provider.Settings;
import android.widget.Toast;

import org.json.JSONObject;

import java.io.IOException;

import de.thu.recipebook.activities.RecipeDetailsActivity;
import de.thu.recipebook.models.RecipeCollection;
import okhttp3.Credentials;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;

public class DeleteRecipeRunnable implements Runnable {
    private RecipeDetailsActivity activity;
    private RecipeCollection recipeCollection;

    public DeleteRecipeRunnable(RecipeDetailsActivity activity) {
        this.activity = activity;
        this.recipeCollection = RecipeCollection.getInstance();
    }

    @Override
    public void run() {
        if (activity != null) {
            String toastText = "Connection to the server couldn't be established.";
            try {
                Response response = executeQuery();
                if (response.isSuccessful()) {
                    recipeCollection.removeRecipeById(activity.getIntent().getStringExtra("id"));
                    toastText = "The recipe was successfully deleted.";
                } else {
                    toastText = new JSONObject(response.body().string()).getString("message");
                }
            } catch (Exception e) {
                e.printStackTrace();
            }

            String finalToastText = toastText;
            activity.runOnUiThread(() -> {
                Toast toast = Toast.makeText(activity, finalToastText, Toast.LENGTH_SHORT);
                toast.show();
            });
            activity.finish();
        }
    }

    private Response executeQuery() throws IOException {
        String id = activity.getIntent().getStringExtra("id");
        String url = "http://10.0.2.2:3000/api/recipes/" + id;
        String credentials = Credentials.basic(Settings.Secure.getString(activity.getContentResolver(),
                Settings.Secure.ANDROID_ID), "");

        OkHttpClient client = new OkHttpClient();
        Request request = new Request.Builder().url(url).delete().header("Authorization", credentials).build();

        Response response = client.newCall(request).execute();
        return response;
    }
}

File: documentation/Exercise 7/Kotlin/src/main/java/de/thu/recipebook/runnables/DeleteRecipeRunnable.kt
package de.thu.recipebook.runnables

import android.provider.Settings
import android.widget.Toast
import de.thu.recipebook.activities.RecipeDetailsActivity
import de.thu.recipebook.models.RecipeCollection
import okhttp3.Credentials.basic
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.Response
import org.json.JSONObject
import java.io.IOException

class DeleteRecipeRunnable(private val activity: RecipeDetailsActivity?) : Runnable {
    private val recipeCollection: RecipeCollection = RecipeCollection.instance!!

    override fun run() {
        if (activity != null) {
            var toastText = "Connection to the server couldn't be established."
            try {
                val response = executeQuery()
                toastText = if (response.isSuccessful) {
                    recipeCollection.removeRecipeById(activity.intent.getStringExtra("id"))
                    "The recipe was successfully deleted."
                } else {
                    JSONObject(response.body!!.string()).getString("message")
                }
            } catch (e: Exception) {
                e.printStackTrace()
            }

            val finalToastText = toastText
            activity.runOnUiThread(Runnable {
                val toast = Toast.makeText(activity, finalToastText, Toast.LENGTH_SHORT)
                toast.show()
            })
            activity.finish()
        }
    }

    @Throws(IOException::class)
    private fun executeQuery(): Response {
        val id = activity!!.intent.getStringExtra("id")
        val url = "http://10.0.2.2:3000/api/recipes/$id"
        val credentials = basic(
            Settings.Secure.getString(
                activity.contentResolver,
                Settings.Secure.ANDROID_ID
            ), ""
        )

        val client = OkHttpClient()
        val request: Request =
            Request.Builder().url(url).delete().header("Authorization", credentials).build()

        return client.newCall(request).execute()
    }
}